### PR TITLE
Lpf 202 dbc code to query static csv sql mapping table

### PR DESCRIPTION
**Pull request by Lukehub1**

### CODE REVIEW

File: src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java [hunk 1]
```diff
@@ -1,53 +1,116 @@
...
 import java.util.List;
+
 @Repository
-public class MappingTableDao {
+@Slf4j
+public class MappingTableDao  implements RowMapper<Object> {
 
     private final List<MappingTableModel> mappingTableObjectList = new ArrayList<>();
+
+//    @Autowired
```
[the remark this answers is not in the document]
Comment (Lukehub1) [reply src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java:23]: Yeh I could delete that now really... I went with a different syntax in the end, as this template one wasn't working

File: src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java [hunk 2]
```diff
@@ -1,53 +1,116 @@
...
+    public List<MappingTableModel> fetchReportList() {
+        ResultSet rslt;
+        int rowNumber = 0;
         mappingTableObjectList.clear(); // Prevent response data accumulating after multiple requests
+        MappingTableModel mappingTableObject;
+
 
-        //fetch data from database
-        //fetchSingleRowOfData()
+        Connection conn = null;
```
Comment (deepujames1234) [inline src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java:87]: Can we avoid calling this each time method is called? Jdbc template can be a better solution

Comment (Lukehub1) [reply src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java:87]: How does JDBC do it then - is it keeping a session open and reusing the same one till you close it or something?

File: src/main/java/uk/gov/laa/pfla/auth/service/services/MappingTableService.java [hunk 3]
```diff
@@ -23,7 +31,8 @@ public MappingTableService(MappingTableDao mappingTableDao) {
         this.mappingTableDao = mappingTableDao;
     }
 
-    public List<ReportListResponse> createReportListResponseList() {
+    public List<ReportListResponse> createReportListResponseList() throws Exception {
+        mappingTableDao.setupDB();
```
Comment (deepujames1234) [inline src/main/java/uk/gov/laa/pfla/auth/service/services/MappingTableService.java:35]: Can we avoid calling setupDB each time?

**Review by deepujames1234 — CHANGES_REQUESTED** [review]
Setup database is called for each call to getReportList
Why we are not using Jdbctemplate?

File: src/main/java/uk/gov/laa/pfla/auth/service/services/MappingTableService.java (hunk 3 above)
Comment (Lukehub1) [reply src/main/java/uk/gov/laa/pfla/auth/service/services/MappingTableService.java:35]: That would make sense, yeh i'll look to see how else to arrange the flow

**Review by deepujames1234 — APPROVED** [review]

**Review by deepujames1234 — APPROVED** [review]

**Pull request merged**